docs/serviceprincipal.md

```diff
@@ -25,12 +25,31 @@ There are several ways to create a Service Principal in Azure Active Directory:
    az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/${SUBSCRIPTION_ID}"
    ```
 
-This will output your `client_id`, `client_secret` (password), `sp_name`, and `tenant`.  The `sp_name` or `client_id` may be used for the `servicePrincipalProfile.servicePrincipalClientId` and the `client_secret` is used for `servicePrincipalProfile.servicePrincipalClientSecret`.
+You can set `SUBSCRIPTION_ID` like this:
 
-Confirm your service principal by opening a new shell and run the following commands substituting in `sp_name`, `client_secret`, and `tenant`:
+```
+SUBSCRIPTION_ID=$(az account list --query "[?name == 'subscription name'].[id]" --out tsv)

```

Comment:
IMO, this subscription_Id section, at least, can be omitted entirely now. We should instead change the above to be `az ad sp create-for-rbac --role=Contributor` which will automatically scope the SP to the subscription.

We should note that the user should ensure they're in the right subscription in the CLI ahead of time (something they already need to know how to do ...) and to make sure they're on the latest version of the `azure-cli`.